### PR TITLE
[Redesign] Update detection operator

### PR DESCRIPTION
**Pull request by objorkman**
* Adds `detection_operator_runner.py` to test operator changes
* Update `detection_operator.py` to erdos redesign operator style

### CODE REVIEW

File: pylot/perception/detection/detection_operator.py [hunk 1]
```diff
@@ -62,53 +53,15 @@ def __init__(self, camera_stream: erdos.ReadStream,
...
-            msg (:py:class:`~pylot.perception.messages.FrameMessage`): Message
-                received.
-            obstacles_stream (:py:class:`erdos.WriteStream`): Stream on which
-                the operator sends
-                :py:class:`~pylot.perception.messages.ObstaclesMessage`
-                messages.
-        """
-        self._logger.debug('@{}: {} received message'.format(
-            msg.timestamp, self.config.name))
+    def on_left_data(self, context: TwoInOneOutContext, data: Any):
```
Comment (pschafhalter) [inline pylot/perception/detection/detection_operator.py:56]:
```suggestion
    def on_left_data(self, context: TwoInOneOutContext, data: CameraFrame):
```

File: pylot/perception/detection/detection_operator.py [hunk 2]
```diff
@@ -171,3 +124,10 @@ def __run_model(self, image_np):
         res_boxes = boxes[0][:num_detections]
         res_scores = scores[0][:num_detections]
         return num_detections, res_boxes, res_scores, res_classes
+
+    def destroy(self):
+        self._logger.warn('destroying {}'.format(self.config.name))
+        # Sending top watermark because the operator is not flowing
+        # watermarks.
+        # self._obstacles_stream.send(
+        #     erdos.WatermarkMessage(erdos.Timestamp(is_top=True)))
```
Comment (pschafhalter) [inline pylot/perception/detection/detection_operator.py:133]: Could you delete this? The new version of ERDOS should automatically send top watermarks after `destroy()` finishes running.

**Review by pschafhalter — APPROVED** [review]
LGTM

File: pylot/perception/detection/detection_operator.py [hunk 3]
```diff
@@ -11,45 +14,34 @@
...
         # Only sets memory growth for flagged GPU
         physical_devices = tf.config.experimental.list_physical_devices('GPU')
-        tf.config.experimental.set_visible_devices(
-            [physical_devices[self._flags.obstacle_detection_gpu_index]],
-            'GPU')
-        tf.config.experimental.set_memory_growth(
-            physical_devices[self._flags.obstacle_detection_gpu_index], True)
+        tf.config.experimental.set_visible_devices([physical_devices[0]],
+                                                   'GPU')
+        tf.config.experimental.set_memory_growth(physical_devices[0], True)
```
Comment (sukritkalra) [inline pylot/perception/detection/detection_operator.py:44]: Can you please revert this to use the flags?

**Review by sukritkalra — CHANGES_REQUESTED** [review]
Can you also please add a TODO to remove the `FrameMessage` when we're done with the port?

**Comment by objorkman** [comment]
Changes made!

File: detection_operator_runner.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            """Invoked when an image is received from the simulator."""
+            game_time = int(simulator_image.timestamp * 1000)
+            timestamp = erdos.Timestamp(coordinates=[game_time])
+            watermark_msg = erdos.WatermarkMessage(timestamp)
+
+            # Ensure that the code executes serially
+            with _lock:
+                msg = None
+                if rgb_camera_setup.camera_type == 'sensor.camera.rgb':
+                    # TODO: Remove FrameMessage when redesign port complete
```
Comment (pschafhalter) [inline detection_operator_runner.py:109]: Could you move this todo above the `FrameMessage` class?
https://github.com/erdos-project/pylot/blob/ac024de97025e0afad7f08202950e436cf3aed36/pylot/perception/messages.py#L14

**Pull request merged**